src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/DataContractJsonSerializer.cs

```diff
@@ -188,7 +193,12 @@ private DataContract RootContract
             {
                 if (_rootContract == null)
                 {
-                    _rootContract = DataContract.GetDataContract(_rootType);
+                    Type contractType = _rootType;
+                    if (_serializationSurrogateProvider != null)
+                    {
+                        contractType = DataContractSurrogateCaller.GetDataContractType(_serializationSurrogateProvider, _rootType);
+                    }
+                    _rootContract = DataContract.GetDataContract(contractType);
```

Comment:
Let's use a consistent single-line pattern like we already use in `DataContractSerializer.RootContract`:
```cs
_rootContract = DataContract.GetDataContract((_serializationSurrogateProvider == null) ? _rootType : GetSurrogatedType(_serializationSurrogateProvider, _rootType));
```